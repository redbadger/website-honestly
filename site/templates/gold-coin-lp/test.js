import React from 'react';
import { render, shallow } from 'enzyme';
import GoldCoinLP from '.';

describe('site/templates/gold-coin-lp', () => {
  const testGoldCoinPage = {
    slug: 'test',
    title: 'test-title',
    subTitle: 'test-subtitle',
    headerImage: 'https://www.example.com/test.jpg',
    headerAlt: 'test-alt',
    duration: 'test-duration',
    price: 'test-price',
    type: 'lunch',
    location: 'test-location',
    whatIsIt: '<p>test what is it</p>',
    whoIsItFor: '<p>test who is it for</p>',
    whatWillYouLearn: '<p>test what will you learn</p>',
    whoWillRun: '<p>test who will run</p>',
    consultants: [
      {
        image: 'https://www.example.com/test-3.jpg',
        name: 'test-consultant',
        role: 'test-role',
        profileUrl: 'https://www.example.com/test-consultant',
      },
    ],
    previews: [
      {
        image: 'https://www.example.com/test-2.jpg',
        title: 'test-title-2',
        subTitle: 'test-subtitle-2',
        url: 'https://www.example.com/',
        duration: 'test-duration-2',
        alt: 'test-alt-2',
      },
    ],
    hubspotForm: {
      portalId: 1111111,
      guid: 'test-guid',
      name: 'test-name',
      cssClass: 'test-css',
      submitText: 'test-submit',
      inlineMessage: 'test-inline',
      formFields: [
        {
          name: 'test-field-name',
          label: 'test-label',
          fieldType: 'lunch',
          required: true,
          enabled: true,
          hidden: false,
          labelHidden: false,
        },
      ],
    },
  };

  it('Renders', () => {
    const shallowRender = props => shallow(<GoldCoinLP {...props} />);
    const component = shallowRender({ ...testGoldCoinPage });
    expect(component).toMatchSnapshot();
  });

  it('should render consultants', () => {
    const goldCoinPage = render(<GoldCoinLP {...testGoldCoinPage} />);

    const consultantList = goldCoinPage.find('.goldCoinLPConsultants').last();
    expect(consultantList.length).toEqual(1);
    expect(consultantList.text()).toMatch(/test-consultant/);
  });

  it('should render previews', () => {
    const goldCoinPage = render(<GoldCoinLP {...testGoldCoinPage} />);

    const previewsList = goldCoinPage.find('.goldCoinLPPreviews').last();
    expect(previewsList.length).toEqual(1);
    expect(previewsList.text()).toMatch(/test-title-2/);
  });
});
