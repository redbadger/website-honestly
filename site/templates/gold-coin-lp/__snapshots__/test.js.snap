// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`site/templates/gold-coin-lp Renders 1`] = `
<div>
  <Social
    altText="test-alt"
    description="Weâ€™re an award winning, independently owned consultancy who believe in doing the right thing and doing the thing right."
    metaImage="https://www.example.com/test.jpg"
    title="test-title | Red Badger"
    url="https://red-badger.com/experience-us/test"
  />
  <div
    className="goldCoinLP"
  >
    <img
      alt="test-alt"
      className="goldCoinLPHeaderImage"
      src="https://www.example.com/test.jpg"
    />
    <div
      className="goldCoinLPContentContainer"
    >
      <div
        className="goldCoinLPContent"
      >
        <div
          className="goldCoinLPDuration"
        >
          TEST-DURATION
        </div>
        <h1
          className="h1"
        >
          test-title
        </h1>
        <h2
          className="h2"
        >
          test-subtitle
        </h2>
        <div
          className="goldCoinLPAtAGlance"
        >
          <p>
            <img
              alt="Price of engagement"
              className="goldCoinLPInfoImg"
              src="price.png"
            />
            test-price
          </p>
          <p>
            <span
              alt="Lunch & learn engagement type icon"
              className="goldCoinLPInfoImg"
            >
              <LunchIcon />
            </span>
            Lunch & learn
          </p>
          <p>
            <span
              className="goldCoinLPInfoLocation"
            >
              <img
                alt="Location of engagement"
                className="goldCoinLPInfoImg"
                src="location.png"
              />
            </span>
            test-location
          </p>
        </div>
        <div
          className="goldCoinLPBody"
        >
          <div
            className="goldCoinLPBodyText"
          >
            <div
              className="goldCoinLPQSection"
            >
              <h3
                className="h3"
              >
                What is it?
              </h3>
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<p>test what is it</p>",
                  }
                }
              />
            </div>
            <div
              className="goldCoinLPQSection"
            >
              <h3
                className="h3"
              >
                Who is this for?
              </h3>
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<p>test who is it for</p>",
                  }
                }
              />
            </div>
            <div
              className="goldCoinLPQSection"
            >
              <h3
                className="h3"
              >
                What will you learn?
              </h3>
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<p>test what will you learn</p>",
                  }
                }
              />
            </div>
            <div
              className="goldCoinLPQSection"
            >
              <h3
                className="h3"
              >
                Who'll run it?
              </h3>
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<p>test who will run</p>",
                  }
                }
              />
              <div
                className="goldCoinLPConsultants"
              >
                <Consultant
                  image="https://www.example.com/test-3.jpg"
                  key="test-consultant"
                  name="test-consultant"
                  profileUrl="https://www.example.com/test-consultant"
                  role="test-role"
                />
              </div>
            </div>
          </div>
          <HubspotForm
            cssClass="test-css goldCoinForm"
            formFields={
              Array [
                Object {
                  "enabled": true,
                  "fieldType": "lunch",
                  "hidden": false,
                  "label": "test-label",
                  "labelHidden": false,
                  "name": "test-field-name",
                  "required": true,
                },
              ]
            }
            guid="test-guid"
            inlineMessage="test-inline"
            name="test-name"
            portalId={1111111}
            submitText="test-submit"
          />
        </div>
      </div>
      <Link
        className="backCTA backCTADesktop"
        to="experienceUs"
      >
        Back to overview
      </Link>
    </div>
    <div
      className="goldCoinLPExplore"
    >
      <div
        className="goldCoinLPExploreContent"
      >
        <h3
          className="h3"
        >
          Thirsty for more?
        </h3>
        <div
          className="goldCoinLPPreviews"
        >
          <PreviewSlice
            alt="test-alt-2"
            duration="test-duration-2"
            image="https://www.example.com/test-2.jpg"
            key="test-title-2"
            subTitle="test-subtitle-2"
            title="test-title-2"
          />
        </div>
      </div>
    </div>
    <Link
      className="backCTA backCTAMobile"
      to="experienceUs"
    >
      Back to overview
    </Link>
  </div>
</div>
`;
